### ISSUE TITLE
Repeating messages on the TTY console

### ISSUE THREAD

**Issue by vanfanel** [bbbradsmith/hatariB#35]
Hi there again, Brad!

I just built the core again from updates cores today, on X86_64 GNU/Linux.
I'm seeing these on the console all time repeating infinitely while the core runs:
```
CPU breakpoints: 0
DSP breakpoints: 0
CPU breakpoints: 0
DSP breakpoints: 0
CPU breakpoints: 0
DSP breakpoints: 0
CPU breakpoints: 0
DSP breakpoints: 0
CPU breakpoints: 0
DSP breakpoints: 0
```

Is there any way to disable that?

Thanks!

**Comment by vanfanel** [comment]
Ah! Seems to be caused by RetroArch's Runahead feature.
Is the core compatible with that? Seems to have no problems, other than hammering the TTY with these messages...

**Comment by bbbradsmith** [comment]
I expect this is related to #28 

Yes, run-ahead is supposed to work, as is netplay, but they rely on seamless savestates, and I haven't yet had any confirmation that savestates work on the linux builds yet. I didn't have a linux environment to work on this issue at the time, but I hope to get to it soon.

**Comment by bbbradsmith** [comment]
Ah, that turned out to be a separate issue but also related to savestates.

These debug messages should no longer appear as of: https://github.com/bbbradsmith/hatariB/commit/c04223e073dfabdc952e1c86c72d158d1c90f75e

Also, savestates should now work.

**Comment by vanfanel** [comment]
@bbbradsmith Debug messages don't appear anymore, indeed!
This core is more and more awesome by the day :)
Should I close this?

**Comment by bbbradsmith** [comment]
Thanks for the confirmation.

**Comment by vanfanel** [comment]
@bbbradsmith Even if as I said there are no more debug messages and Runahead seems to work, CPU usage increments from 10% without Runahead to ~70% with Runahead (only 1 core is used).

Is that supposed to be right? Other cores don't show such a massive increase in CPU usage with Runahead.

**Comment by bbbradsmith** [comment]
Savestates are a very heavy operation in Hatari. I made some optomizations to reduce it to a fraction of what it was, but it's still relatively high. Using run-ahead means taking and restoring savestates constantly, so yes that seems expected.

**Comment by vanfanel** [comment]
@bbbradsmith thanks for taking the time to answer. Yes, savestates can be heavy on some emulators, so I perfectly understand. I also thank you for trying to optimize It!

**Comment by bbbradsmith** [comment]
It's actually one of the places I've put the most work. Less for run-ahead and more just for netplay. I got it down enough that netplay seems to work reasonably on my current mid-range PC, but Hatari's savestate code design was based around being thorough, with no speed considerations at all. It really expects to just fully restart the entire system each time, which for the stand-alone version doing savestates in response to user input is still "instant", but as this libretro core for run-ahead or netplay it's very significant. There's always more optimization that could be done, if one can find the time, but I got the most major ones out of the way for now.
